src/components/ComponentParts/RedButton.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+
+interface Props {
+  link: string;
+  buttonMessage: string;
+  buttonType: 'submit' | 'reset' | 'button' | 'menu';
```

Comment:
@Muniox tutaj temat do Ciebie. Jak rozumiem, mamy w eslincie ustawione, że nie możemy robić czegoś w stylu `{...rest}`.
Niestety, ale takie założenie sprawia, że tworzenie prostych, podstawowych komponentów wymaga by dodawać im wszystkie potrzebne propsy, przez co te komponenty od razu nie są proste.

Podam przykład z mojego innego projektu (może on nie jest idealny, ale spójrz jak potem ułatwia korzystanie z takich gotowych komponentów):
```js
import React from 'react';

interface Props extends React.InputHTMLAttributes<HTMLInputElement> {
  label: string;
  rest?: React.InputHTMLAttributes<HTMLInputElement>;
}

export const FormInput = ({ name, label, ...rest }: Props) => {
  return (
    <div className="flex flex-col-reverse">
      <input
        id={name}
        name={name}
        className="bg-transparent rounded border-2 dark:border-dark-subtle border-light-subtle dark:focus:border-white focus:border-primary p-1 w-full text-lg dark:text-white text-secondary outline-none peer"
        {...rest}
      />
      <label
        htmlFor={name}
        className="font-semibold dark:text-dark-subtle text-light-subtle dark:peer-focus:text-white peer-focus:text-primary transition self-start"
      >
        {label}
      </label>
    </div>
  );
};
```
Potem wykorzystujemy go jak chcemy, gdzie chcemy. A jak chcemy coś w propsach dodatkowego dodać, to nie musimy poprawiać nic w naszym ustalonym komponencie, tylko dodajemy:

```js
<FormInput
            value={email}
            onChange={handleChange}
            name="email"
            placeholder="joe@example.com"
            label="Email"
            type="text"
          />
```
Prosiłbym o ponowne przemyślenie tej decyzji :)
@Annshka @joannaFiolka @MaJan433 @EckhartPL prosiłbym o Waszą opinię :)

Reply:
Trudno mi się jednoznacznie wypowiedzieć. Destrukturyzacja ogólnie jest na plus, ale fakt, to utrudnia pracę z kodem. Tyle, że wiele ustawień ESlintowych takie jest. Wydaje mi się, że wiele z nich mogłoby być zmienione.

Reply:
Jeśli to nie jest jakaś duża sprawa, uważam żeby lepiej dopuścić przypisanie destrukturyzacji. Szczególnie przyda się to gdy chcemy pominąć jakieś właściwości. Ten problem pojawia się też u mnie przy okazji używania react-table hook. Przy obecnych ustawieniach eslinta nie będę w stanie raczej przerobić komponentu tablicowego z uwagi na brakujący czas...